### DISCUSSION TITLE
JS error on iOS chrome versus PC Chrome.

### DISCUSSION THREAD

**Discussion by raventechsaigon** [Budibase/budibase#15190] [Help]
**Checklist**
- [X ] I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**

- Cloud
  - Tenant ID: <raventech> 

**Describe the bug**
For the life of me I can not figure this error out

On Chrome PC...everthing works as expected
On Chrome on Iphone15
Get Javascript error

phone browser
![image](https://github.com/user-attachments/assets/42dde658-6c6d-4220-88c7-09f3aa23d08f)

pc browser
![image](https://github.com/user-attachments/assets/3a27617d-708b-4eda-83a1-b2fa1715cd35)

JS Script

var starttime =$("State.HOURLYMIN");   //state
var d = new Date(starttime);
sd=d.toISOString().slice(0, 19).replace('T', ' ');
var f = new Date(d.getTime()); 
d.setMinutes(d.getMinutes() + 60);
ed=d.toISOString().slice(0, 19).replace('T', ' ');

var today = new Date($("State.TIMENOW"));   //state
timenow = today.toISOString().slice(0, 19).replace('T', ' ');
mindiff =  $("State.HOURLYDELTA")+ " minutes until matchplay!";
out= 'START-' +sd + '   END-'+ed  + "\n   NOW-"+timenow+'   '+mindiff ;

return out;

[Match Play-export-1734205103910.tar.gz](https://github.com/user-attachments/files/18137434/Match.Play-export-1734205103910.tar.gz)

**Comment by linear** [comment]
<p><a href="https://linear.app/budibase/issue/BUDI-8924/js-error-on-ios-chrome-versus-pc-chrome">BUDI-8924 JS error on iOS chrome versus PC Chrome.</a></p>

**Comment by mike12345567** [comment]
Hi @raventechsaigon i've converted this to a discussion as this is not a bug with Budibase itself and more a difference in how the JS engine in Chrome works on iOS vs Windows (perhaps a bug there).

**Reply by raventechsaigon** [reply]
I thought JS was interpreted on server side.
NO JS code is executed by the browser.

The code here is processing states set on server side

**Reply by mike12345567** [reply]
Hi @raventechsaigon - if the JS is set in the design section then it is processed in the browser. The only JS which is processed server-side is automation scripts and formulas - all other JS will be executed in the browser.
